specification/consumption/Consumption.Management/routes.tsp

```diff
@@ -0,0 +1,935 @@
+// FIXME: Operations in this file are not detected as a resource operation, please confirm the conversion result manually
+
+import "@azure-tools/typespec-azure-core";
+import "@typespec/rest";
+import "./models.tsp";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using Azure.ResourceManager;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.Consumption;
+
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-interface-requires-decorator" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+interface PriceSheetOperationGroup {
+  /**
+   * Generates the pricesheet for the provided billing period asynchronously based on the enrollment id
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @externalDocs("https://docs.microsoft.com/en-us/rest/api/consumption/")
+  @autoRoute
+  @action("pricesheets/download")
+  @post
+  @armResourceCollectionAction
+  downloadByBillingAccountPeriod(
+    ...ApiVersionParameter,
+
+    /** the provider namespace */
+    @path
+    @segment("providers")
+    @key
+    providerNamespace: "Microsoft.Billing",
+
+    /**
+     * BillingAccount ID
+     */
+    @path
+    @segment("billingAccounts")
+    billingAccountId: string,
+
+    /**
+     * Billing Period Name.
+     */
+    @path
+    @segment("billingPeriods")
+    billingPeriodName: string,
+
+    ...Azure.ResourceManager.Legacy.Provider,
+  ): OperationStatus | ArmAcceptedLroResponse<LroHeaders = ArmLroLocationHeader<FinalResult = OperationStatus>> | CommonTypes.ErrorResponse;
+}
```

Comment:
I think we should be able to use `ExternalResource` and define an extension action here.